src/Security/Authorization/Policy/src/AuthorizationEndpointConventionBuilderExtensions.cs

```diff
@@ -79,6 +79,66 @@ public static TBuilder RequireAuthorization<TBuilder>(this TBuilder builder, par
         return builder;
     }
 
+    /// <summary>
+    /// Adds an authorization policy to the endpoint(s).
+    /// </summary>
+    /// <param name="builder">The endpoint convention builder.</param>
+    /// <param name="policy">The <see cref="AuthorizationPolicy"/> policy.</param>
+    /// <returns>The original convention builder parameter.</returns>
+    public static TBuilder RequireAuthorization<TBuilder>(this TBuilder builder, AuthorizationPolicy policy)
+        where TBuilder : IEndpointConventionBuilder
+    {
+        if (builder == null)
+        {
+            throw new ArgumentNullException(nameof(builder));
+        }
+
+        if (policy == null)
+        {
+            throw new ArgumentNullException(nameof(policy));
+        }
+
+        // REVIEW: should we check if there already is any authorize attributes first?
+        builder.Add(endpointBuilder =>
+        {
+            endpointBuilder.Metadata.Add(new AuthorizeAttribute());
```

Comment:
Should we check if any IAuthorizeData is already in the metadata so we don't add extra empty `[Authorize]`s?

Reply:
Yes, but I'm not sure where. Can we do it in this callback? We need an "only add metadata if not already added"

Reply:
I just added a Any query inline here to prevent dupes, does that work?